### PR TITLE
Use encoded URL when redirecting user notebooks

### PR DESCRIPTION
**Pull request by yuvipanda**
Otherwise it breaks for usernames that have url unsafe
characters.

### CODE REVIEW

File: jupyterhub/handlers/pages.py [hunk 1]
```diff
@@ -75,9 +75,11 @@ def get(self):
             self.finish(self._render_form())
         else:
             # not running, no form. Trigger spawn.
-            url = url_path_join(self.base_url, 'user', user.name)
+            # Creating the URL manually since the server does not
+            # exist yet
+            url = url_path_join(self.base_url, 'user', quote(user.name))
```
Comment (minrk) [inline jupyterhub/handlers/pages.py]: This should be user.escaped_name

File: share/jupyter/hub/templates/home.html [hunk 2]
```diff
@@ -10,7 +10,7 @@
       {% endif %}
       <a id="start" class="btn btn-lg btn-success"
       {% if user.running %}
-        href="{{base_url}}user/{{user.name}}/"
+        href="{{ user.server.base_url }}"
```
Comment (minrk) [inline share/jupyter/hub/templates/home.html]: This should be `user.url`

**Comment by minrk** [comment]
Thanks for digging into this! The only reason we don't use `user.url` everywhere is that it is retrieved from user.server, which is undefined unless the user is running. But there's no actual reason for this restriction (there may be, once we have multiple servers), so after this fix, I think I'll make it so `user.url` is always computable, and make sure we never build user URLs without it.

File: share/jupyter/hub/templates/home.html (hunk 2 above)
Comment (yuvipanda) [inline share/jupyter/hub/templates/home.html]: done!

On Wed, Mar 9, 2016 at 12:18 AM, Min RK <notifications@github.com> wrote:

> In share/jupyter/hub/templates/home.html
> <https://github.com/jupyter/jupyterhub/pull/472#discussion_r55484142>:
>
> > @@ -10,7 +10,7 @@
> >        {% endif %}
> >        <a id="start" class="btn btn-lg btn-success"
> >        {% if user.running %}
> > -        href="{{base_url}}user/{{user.name}}/"
> > +        href="{{ user.server.base_url }}"
>
> This should be user.url
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyterhub/pull/472/files#r55484142>.
>



-- 
Yuvi Panda T
http://yuvi.in/blog

**Comment by yuvipanda** [comment]
@minrk yw! And thanks for merging! It looks like the commit with the changes you suggested got lost somehow - maybe my network died before I pushed it, and hence the merge didn't have the changes you suggested. Want me to put that in as a different branch? But I see that you're changing it anyway with #474 so maybe it is moot :)

**Comment by minrk** [comment]
I might have merged before you got to it. Sorry! I did the user.url change in #474, which should make this harder to mess up in the future.

**Pull request merged**